### PR TITLE
add prometheus to dev cluster

### LINKED ISSUE #437: add prometheus to AKDC
- add prometheus and grafana to AKDC to view prometheus data

### LINKED ISSUE #441: add grafana to AKDC
- add grafana to dev cluster for prometheus dashboards

### PR DESCRIPTION
**Pull request by bartr**
# Don Not Merge
- need to fully test design review updates

# Type of PR

- [x] Documentation changes
- [ ] Code changes
- [ ] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes
- [x] yaml changes

## Purpose of PR
- add setup instructions for prometheus / grafana on dev cluster

## Validation

- [ ] Unit tests updated and ran successfully
- [x] Update documentation or issue referenced above

## Issues Closed or Referenced

- Closes #437 
- Closes #441 
- Closes #442

### CODE REVIEW

**Review by aflinchb — APPROVED** [review]
Approved w/ a question: should the other yaml files in DevCluster/loderunner be updated for prometheus? (i.e. baseline-memory.yaml, benchmark-cosmos.yaml, etc.)

**Review by jofultz — APPROVED** [review]
What's needed for the prometheus ConfigMap is new to me, but is gtg as best as I can tell.

Also, I'm wondering if we should set the zone and region flags from a configMap and then grab the values with envFrom.

File: IaC/DevCluster/prometheus/1-clusterRole.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  - watch
+
+- nonResourceURLs:
+  - /metrics
+  verbs:
+  - get
+
+---
+
+apiVersion: rbac.authorization.k8s.io/v1beta1
```
Comment (AAkindele) [inline IaC/DevCluster/prometheus/1-clusterRole.yaml:35]: I got a deprecation warning here.

```
Warning: rbac.authorization.k8s.io/v1beta1 ClusterRoleBinding is deprecated in v1.17+, unavailable in v1.22+; use rbac.authorization.k8s.io/v1 ClusterRoleBinding
```
```suggestion
apiVersion: rbac.authorization.k8s.io/v1
```

**Review by AAkindele — APPROVED** [review]

File: IaC/DevCluster/prometheus/1-clusterRole.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  name: prometheus
+
+roleRef:
+  name: prometheus
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+
+subjects:
+- kind: ServiceAccount
+  name: default
```
[the remark this answers is not in the document]
Comment (bartr) [reply IaC/DevCluster/prometheus/1-clusterRole.yaml:47]: good catch - I'll add that

File: IaC/DevCluster/prometheus/1-clusterRole.yaml (hunk 1 above)
Comment (bartr) [reply IaC/DevCluster/prometheus/1-clusterRole.yaml:35]: I thought I fixed that ... updated

**Pull request merged**